### DISCUSSION TITLE
npm run build fail

### DISCUSSION THREAD

**Discussion by kenshepherd58** [tailwindlabs/tailwindcss#2620] [Help] [answered]
Installed Tailwind as per Adam's course. Got to the "build" step (4:30 in video) in the package.json file and tried "npm run build" in Terminal which threw the following errors:

> tailwind-dev-learning@1.0.0 build /Users/kshepherd/Documents/tailwind/tailwind-dev-learning
> postcss css/tailwind.css -o public/build/tailwind.css

Error: PostCSS plugin autoprefixer requires PostCSS 8.
Migration guide for end-users:
https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users
    at Processor.normalize (/Users/kshepherd/Documents/tailwind/tailwind-dev-learning/node_modules/postcss/lib/processor.js:153:15)
    at new Processor (/Users/kshepherd/Documents/tailwind/tailwind-dev-learning/node_modules/postcss/lib/processor.js:56:25)
    at postcss (/Users/kshepherd/Documents/tailwind/tailwind-dev-learning/node_modules/postcss/lib/postcss.js:55:10)
    at /Users/kshepherd/Documents/tailwind/tailwind-dev-learning/node_modules/postcss-cli/index.js:217:14
    at async Promise.all (index 0)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! tailwind-dev-learning@1.0.0 build: `postcss css/tailwind.css -o public/build/tailwind.css`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the tailwind-dev-learning@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kshepherd/.npm/_logs/2020-10-19T18_27_05_950Z-debug.log

-------------- end of errors

Up to this point, everything installed/worked perfectly.

Do I need an updated version of PostCSS? If so, how do I update?

Current configuration...
+ autoprefixer@10.0.1
+ postcss-cli@8.1.0
+ tailwindcss@1.9.5

Thanks

**Comment by adamwathan** [comment, marked as answer]
If you run `npm install postcss` as well that should fix it. Some things have changed with PostCSS 8 that unfortunately make the video instructions out of date.

**Reply by kenshepherd58** [reply]
Worked. Thank you!

Love the product. I'm a designer considering using Tailwind CSS to replace Adobe XD and outsourced CSS markup. Utilizing Tailwind I can complete all the HTML/CSS UI myself and hand this off to the Javascript developers to complete each project. Should minimize our design/development cycle and provide a more consistent/standardize UI look and feel. Buying the early access bundle today. Look forward to learning Tailwind and the current and future components you come up with.

**Reply by IslamicProgrammer** [reply]
In my case this is not working :(

**Comment by petekal1323** [comment]
I have the same problem and even ran npm install postcss

C:\Users\pkalinowski\Desktop\React-Projects\learning-react (master)
λ npm run build:css

> learning-react@0.1.0 build:css C:\Users\pkalinowski\Desktop\React-Projects\learning-react
> postcss src/index.css -o src/tailwind.css

Error: Cannot find module 'autoprefixer'
Require stack:
- C:\Users\pkalinowski\Desktop\React-Projects\learning-react\postcss.config.js
- C:\Users\pkalinowski\Desktop\React-Projects\learning-react\node_modules\postcss-load-config\node_modules\cosmiconfig\dist\loaders.js
- C:\Users\pkalinowski\Desktop\React-Projects\learning-react\node_modules\postcss-load-config\node_modules\cosmiconfig\dist\ExplorerBase.js
- C:\Users\pkalinowski\Desktop\React-Projects\learning-react\node_modules\postcss-load-config\node_modules\cosmiconfig\dist\Explorer.js
- C:\Users\pkalinowski\Desktop\React-Projects\learning-react\node_modules\postcss-load-config\node_modules\cosmiconfig\dist\index.js
- C:\Users\pkalinowski\Desktop\React-Projects\learning-react\node_modules\postcss-load-config\src\index.js
- C:\Users\pkalinowski\Desktop\React-Projects\learning-react\node_modules\postcss-cli\index.js
- C:\Users\pkalinowski\Desktop\React-Projects\learning-react\node_modules\postcss-cli\bin\postcss
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (C:\Users\pkalinowski\Desktop\React-Projects\learning-react\postcss.config.js:4:7)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Module.load (internal/modules/cjs/loader.js:985:32)
    at Function.Module._load (internal/modules/cjs/loader.js:878:14)
    at Module.require (internal/modules/cjs/loader.js:1025:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\pkalinowski\\Desktop\\React-Projects\\learning-react\\postcss.config.js',
    'C:\\Users\\pkalinowski\\Desktop\\React-Projects\\learning-react\\node_modules\\postcss-load-config\\node_modules\\cosmiconfig\\dist\\loaders.js',
    'C:\\Users\\pkalinowski\\Desktop\\React-Projects\\learning-react\\node_modules\\postcss-load-config\\node_modules\\cosmiconfig\\dist\\ExplorerBase.js',
    'C:\\Users\\pkalinowski\\Desktop\\React-Projects\\learning-react\\node_modules\\postcss-load-config\\node_modules\\cosmiconfig\\dist\\Explorer.js',
    'C:\\Users\\pkalinowski\\Desktop\\React-Projects\\learning-react\\node_modules\\postcss-load-config\\node_modules\\cosmiconfig\\dist\\index.js',
    'C:\\Users\\pkalinowski\\Desktop\\React-Projects\\learning-react\\node_modules\\postcss-load-config\\src\\index.js',
    'C:\\Users\\pkalinowski\\Desktop\\React-Projects\\learning-react\\node_modules\\postcss-cli\\index.js',
    'C:\\Users\\pkalinowski\\Desktop\\React-Projects\\learning-react\\node_modules\\postcss-cli\\bin\\postcss'
  ]
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! learning-react@0.1.0 build:css: `postcss src/index.css -o src/tailwind.css`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the learning-react@0.1.0 build:css script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\pkalinowski\AppData\Roaming\npm-cache\_logs\2020-10-27T16_49_21_485Z-debug.log

**Reply by adamwathan** [reply]
Looks like you forgot to install autoprefixer based on the error.

**Comment by SpyShare** [comment]
Try Write `npm i -D postcss-cli`
My problem is solved
